src/domain/service/note.ts

```diff
@@ -441,4 +449,40 @@ export default class NoteService {
 
     return noteHistoryPublic;
   }
+
+  /**
+   * Get note parent structure recursively by note id and user id
+   * and check if user has access to the parent note.
+   * @param noteId - id of the note to get parent structure
+   * @param userId - id of the user that is requesting the parent structure
+   * @returns - array of notes that are parent structure of the note
+   */
+  public async getNoteParentStructure(noteId: NoteInternalId, userId: number | null): Promise<Array<{ noteId: Note['publicId']; content: Note['content'] }>> {
+    const noteUserAccess = await this.teamRepository.getTeamMemberByNoteAndUserId(userId as NoteInternalId, noteId);
+
+    if (userId === null || noteUserAccess === null) {
+      const nextNoteId = await this.getParentNoteIdByNoteId(noteId);
+
+      if (nextNoteId === null || userId === null) {
+        return [];
+      }
+
+      return [...await this.getNoteParentStructure(nextNoteId, userId)];
+    }
+
+    const parentNoteId = await this.getParentNoteIdByNoteId(noteId);
+
+    const noteData = await this.getNoteById(noteId);
+
+    const noteObjectInfo = {
+      noteId: noteData.publicId,
+      content: noteData.content,
+    };
+
+    if (parentNoteId === null) {
+      return [noteObjectInfo];
+    }
```

Comment:
i suggest moving this method to teams model level and using teamsModel with included noteModel